lib/chewy/search/parameters.rb

```diff
@@ -3,16 +3,30 @@
 
 module Chewy
   module Search
+    # This class is basically a compoung storage of the request
+    # parameter storages. It incapsulates some storage-collection-handling
+    # logic.
+    #
+    # @see Chewy::Search::Request#parameters
     class Parameters
+      # Deafult storage classes warehouse. It is probably possible to
+      # add your own classes here if necessary, but I'm not sure it will work.
+      #
+      # @return [{Symbol => Chewy::Query::Parameters::Value}]
       def self.storages
         @storages ||= Hash.new do |hash, name|
           hash[name] = "Chewy::Search::Parameters::#{name.to_s.camelize}".constantize
         end
       end
 
+      # @return [{Symbol => Chewy::Query::Parameters::Value}]
       attr_accessor :storages
       delegate :[], :[]=, to: :storages
 
+      # Accepts a hash of initial values as basic subobjects or
```

Comment:
What are "subobjects"? Do you mean instances of subclasses? That way it would be much clearer.